NomadUI/Widgets/PluginBrowserPanel.cpp

```
@@ -211,29 +223,87 @@ bool PluginBrowserPanel::onMouseEvent(const NUIMouseEvent& event) {
         NUIRect scanBtn = {bounds.x + bounds.width - 80, bounds.y + 8, 64, 24};
         if (mx >= scanBtn.x && mx < scanBtn.x + scanBtn.width &&
             my >= scanBtn.y && my < scanBtn.y + scanBtn.height) {
-            if (m_onScanRequested) {
+            if (!m_scanning && m_onScanRequested) {
                 m_onScanRequested();
             }
             return true;
         }
     }
+    // Handle Release (Click)
+    else if (event.released && event.button == NUIMouseButton::Left) {
```

Clang (14.0.6)

[note] +1, nesting level increased to 1

(clang)